src/Repository/FindByChannelWithGatewayConfigTrait.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CommerceWeavers\SyliusTpayPlugin\Repository;
+
+use Doctrine\ORM\EntityRepository;
+use Sylius\Component\Channel\Model\ChannelInterface;
+
+/**
+ * @mixin EntityRepository
+ */
+trait FindByChannelWithGatewayConfigTrait
+{
+    public function findByChannelAndGatewayConfigNameWithGatewayConfig(ChannelInterface $channel, string $gatewayConfigName): array
+    {
+        return $this->createQueryBuilder('o')
+            ->addSelect('gatewayConfig')
+            ->leftJoin('o.gatewayConfig', 'gatewayConfig')
```

Comment:
```suggestion
            ->innerJoin('o.gatewayConfig', 'gatewayConfig', Join::WITH, 'gatewayConfig.gatewayName = :gatewayName')
```

and then we can remove the line `->andWhere('gatewayConfig.gatewayName = :gatewayName')`

Reply:
or you can leave where statement as it is but I'd still replace a left join with an inner join.